src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4696,19 +4748,22 @@ private ParameterSyntax ParseParameter()
             }
 
             var type = this.ParseType(mode: ParseTypeMode.Parameter);
-            SyntaxToken identifier;
 
+            SyntaxToken? identifier;
             if (this.CurrentToken.Kind == SyntaxKind.IdentifierToken && IsCurrentTokenWhereOfConstraintClause())
             {
                 identifier = this.AddError(CreateMissingIdentifierToken(), ErrorCode.ERR_IdentifierExpected);
```

Comment:
>identifier = this.AddError(CreateMissingIdentifierToken(), ErrorCode.ERR_IdentifierExpected); [](http://example.com/codeflow?start=16&length=93)

It feels like this error recovery might not be appropriate when identifier is optional. Specifically, we probably do not want to create a missing identifier.